python_modules/dagster/dagster/_core/definitions/schedule_definition.py

```diff
@@ -530,7 +532,7 @@ def with_updated_job(self, new_job: ExecutableDefinition) -> "ScheduleDefinition
         return ScheduleDefinition.dagster_internal_init(
             name=self.name,
             cron_schedule=self._cron_schedule,
-            job_name=self.job_name,
+            job_name=None,
```

Comment:
is this on purpose? If so needs a comment for sure

Reply:
yes, I've added a comment. I added logic to make sure you can't simultaneously pass `job_name` and `job` (which should've been there anyway), and in this case the job name will be sourced from the new job in this method